pkg/deploy/deploy.go

```diff
@@ -359,6 +358,9 @@ func getResource(ctx context.Context, cli client.Client, obj *unstructured.Unstr
 	found.SetGroupVersionKind(obj.GroupVersionKind())
 	err := cli.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, found)
 	if err != nil {
+		if strings.Contains(err.Error(), "no matches for kind") {
```

Comment:
Won't the same work? https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/pkg/upgrade/upgrade.go#L299